log/logger.go

```diff
@@ -471,3 +477,204 @@ func parseSyslogAddress(location string) *syslogAddress {
 
 	return nil
 }
+
+const (
+	compressSuffix = ".gz"
+)
+
+// millRunOnce performs compression and removal of stale log files.
+// Log files are compressed if enabled via configuration and old log
+// files are removed, keeping at most l.MaxBackups files, as long as
+// none of them are older than MaxAge.
+func (l *Logger) millRunOnce() error {
+	files, err := l.oldLogFiles()
+	if err != nil {
+		return err
+	}
+
+	compress, remove := l.screeningCompressFile(files)
+
+	for _, f := range remove {
+		_ = os.Remove(filepath.Join(l.dir(), f.FileName))
+	}
+	var wg sync.WaitGroup
+	for _, f := range compress {
+		var fnCompress, fileName string
+		fileName = f.FileName
+		wg.Add(1)
+		fnCompress, err = l.findCompressFile(fileName)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "logger %s findCompressFile , error: %v", l.output, err)
+			return err
+		}
+		go func(fnCompress, fileName string, wg *sync.WaitGroup) {
+			wg.Done()
```

Comment:
```suggestion
			defer wg.Done()
```
这样更好？确保 `compressLogFile` 执行完了？